contracts/src/v0.8/dev/VRFCoordinatorV2.sol

```diff
@@ -0,0 +1,435 @@
+pragma solidity ^0.8.0;
+
+import "../interfaces/LinkTokenInterface.sol";
+import "../interfaces/BlockHashStoreInterface.sol";
+import "../interfaces/AggregatorV3Interface.sol";
+
+import "./VRF.sol";
+import "./ConfirmedOwner.sol";
+
+contract VRFCoordinatorV2 is VRF, ConfirmedOwner {
+
+    LinkTokenInterface public immutable LINK;
+    AggregatorV3Interface public immutable LINK_ETH_FEED;
+    AggregatorV3Interface public immutable FAST_GAS_FEED;
+    BlockHashStoreInterface public immutable BLOCKHASH_STORE;
+
+    event SubscriptionCreated(uint64 subId, address owner, address[] consumers);
+    event SubscriptionFundsAdded(uint64 subId, uint256 oldBalance, uint256 newBalance);
+    event SubscriptionConsumersUpdated(uint64 subId, address[] oldConsumers, address[] newConsumers);
+    event SubscriptionFundsWithdrawn(uint64 subId, uint256 oldBalance, uint256 newBalance);
+    event SubscriptionCanceled(uint64 subId);
+    uint64 private currentSubId;
+    struct Subscription {
+        uint256 balance; // Common balance used for all consumer requests.
+        address owner; // Owner can fund/withdraw/cancel the sub
+        address[] consumers; // List of addresses which can consume using this subscription.
+    }
+    mapping(uint64 /* subId */ => Subscription /* subscription */) private s_subscriptions;
```

Comment:
Events, mappings, and storage variables seem to be mixed together. Is there a pattern to it? I tend to prefer grouping each type together, although the Offchain Aggregator instead declares things close to where they are used. Any specific pattern here?

Reply:
I was grouping them by concept - i.e. storage, events, structdefs all related to subscriptions are in one place etc.

Reply:
Maybe its more clear now order is (errors, structs, storage, events) for subscriptions, randomness req/response, then global config